src/main/java/spark/embeddedserver/EmbeddedServer.java

```diff
@@ -38,8 +38,10 @@
      * @param maxThreads              - max nbr of threads.
      * @param minThreads              - min nbr of threads.
      * @param threadIdleTimeoutMillis - idle timeout (ms).
+     *
+     * @return The port number the server was launched on.
      */
-    void ignite(String host,
+    int ignite(String host,
```

Comment:
This needs to return the port number or else the Service object wont know that the port number has been changed and will keep returning the default port (4567).